sdk/nodejs/provisioning/provisioner.ts

```diff
@@ -0,0 +1,29 @@
+import { EngineConn } from './engineconn.js';
+import { DockerImage } from './docker-provision/image.js';
+
+type ProvisionerFunc = (u: URL) => EngineConn;
+
+const provisioners: Record<string, ProvisionerFunc> = {
+	'docker-image': (u: URL) => new DockerImage(u),
+};
+
+/**
+ * getProvisioner returns a ready to use Engine connector.
+ * This method parse the given host to find out which kind
+ * of provisioner shall be returned.
+ *
+ * It supports the following provisioners:
+ * - docker-image
```

Comment:
as a follow up we will want `http` too so we can run tests via dagger-in-dagger. Doesn't need to block this PR though

Reply:
Issue https://github.com/dagger/dagger/issues/3834 created to keep track of this.